pkg/transport/transport.go

```diff
@@ -30,15 +31,26 @@ type Client interface {
 
 // Transport interface structure
 type Transport struct {
-	tlsOpts   *tlsopts.Options
-	observers []Observer
+	tlsOpts        *tlsopts.Options
+	observers      []Observer
+	requestTimeout time.Duration
 }
 
-// NewClient returns a newly instantiated Transport Client
+// NewClient returns a transport client with a specified timeout for requests
 func NewClient(tlsOpts *tlsopts.Options, obs ...Observer) Client {
 	return &Transport{
```

Comment:
usually here it's nicer to call the more complex func here with the arguments, it avoids bugs by forgetting to add fields

Reply:
oh like, return `NewClientWithTimeout` but pass in the default?

Reply:
yes